mirror_node_contract_query.go

```diff
@@ -0,0 +1,318 @@
+package hiero
+
+import (
+	"bytes"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strconv"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+// SPDX-License-Identifier: Apache-2.0
+
+// mirrorNodeContractQuery returns a result from EVM execution such as cost-free execution of read-only smart contract
+// queries, gas estimation, and transient simulation of read-write operations.
+type mirrorNodeContractQuery struct {
+	// The contract we are sending the transaction to
+	contractID         *ContractID
+	contractEvmAddress *string
+	// The account we are sending the transaction from
+	sender           *AccountID
+	senderEvmAddress *string
+	// The transaction callData
+	callData []byte
+	// The amount we are sending to the contract
+	value *int64
+	// The gas limit
+	gasLimit *int64
+	// The gas price
+	gasPrice *int64
+	// The block number for the simulation
+	blockNumber *int64
+}
+
+// setContractID sets the contract instance to call
+func (query *mirrorNodeContractQuery) setContractID(contractID ContractID) {
```

Comment:
nit: Would also be good to differentiate between  the `query` naming here as we also have `query` in the transaction flow just for ease for code navigation and etc.

Reply:
named `mirrorNodeContractQuery` for clarity